crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/validator/ProsecutionConcludedValidator.java

```diff
@@ -22,12 +39,37 @@ public void validateRequestObject(ProsecutionConcluded prosecutionConcluded) {
             throw new ValidationException(PAYLOAD_IS_NOT_AVAILABLE_OR_NULL);
     }
 
-    public Optional<Void> validateOuCode(String ouCode) {
-
-        if (!isBlank(ouCode)) {
-            return Optional.empty();
-        } else {
+    public String validateOuCode(String ouCode) {
+        if (isBlank(ouCode)) {
             throw new ValidationException(OU_CODE_IS_MISSING);
         }
+
+        try {
+            return crownCourtCodeHelper.getCode(ouCode);
+        }
+        catch (MAATCourtDataException ex) {
+            throw new ValidationException(OU_CODE_LOOKUP_FAILED);
+        }
+    }
+
+    public void validateMaatId(String message) {
+        if (StringUtils.isEmpty(message)) {
+            throw new ValidationException(MAAT_ID_NOT_SUPPLIED);
+        }
+
+        try {
+            JsonObject msgObject = JsonParser.parseString(message).getAsJsonObject();
+            msgObject.get("maatId").getAsInt();
+        } catch (NumberFormatException ex) {
+            throw new ValidationException(MAAT_ID_FORMAT_INCORRECT);
+        }
+    }
+
+    public void validateMagsCourtOutcomeExists(RepOrderDTO repOrderDTO, List<RepOrderCCOutcomeDTO> repOrderOutcomeDtoList) {
+        if (repOrderDTO != null && repOrderDTO.getMagsOutcome() == null &&
+            CollectionUtils.isNotEmpty(repOrderOutcomeDtoList) &&
+            !CaseType.APPEAL_CC.getCaseType().equalsIgnoreCase(repOrderDTO.getCatyCaseType())) {
+            throw new ValidationException(CANNOT_HAVE_CROWN_COURT_OUTCOME_WITHOUT_MAGS_COURT_OUTCOME);
+        }
```

Comment:
Not too sure about this validation. There will never be a queue message that pertains to an `APPEAL_CC` case - this will be added in the future, but I don't think we should pre-empt that work.

The `repOrderOutcomeDtoList` will always be empty if there is no mags outcome, so there's no need to check the list, I'd just do a simple check on the mags outcome being null.

Reply:
I wasn't too sure about this either to be fair. This is taken from the existing validation in _CrownCourtDetailsValidator_ that throws the same exception.

Noted, will check solely whether the magistrates court outcome is null or empty 👍

Reply:
Check has now been simplified as suggested